### DISCUSSION TITLE
AttributeError: 'Resize' object has no attribute 'transforms'. Did you mean: '_transform'?

### DISCUSSION THREAD

**Discussion by stevensagaert** [open-edge-platform/anomalib#2242] [Q&A] [answered]
I get this error when I insert a Resize transformation in the datamodule. The model is EfficientAD.
I suspect this is due to a torch/torchvision incompatibility. I've downgraded torch to 2.2.2+cu121 and even 1.8 but I still get the error. What's the solution?

**Comment by alexriedel1** [comment]
Please show your code :)

**Comment by stevensagaert** [comment]
from anomalib.data import Folder
from anomalib.data.utils import ValSplitMode
#from anomalib.models import Patchcore
from anomalib.models import EfficientAd
from anomalib.engine import Engine
from anomalib import TaskType
from anomalib.loggers import AnomalibTensorBoardLogger
from anomalib.callbacks import ModelCheckpoint
from lightning.pytorch.callbacks import EarlyStopping
from anomalib.deploy import ExportType
from torchvision.transforms.v2 import Resize,Compose
resize_transf = Resize((512, 512))

datamodule = Folder(
    name="ABC",
    root="data",
    normal_dir="normal",
    abnormal_dir="abnormal",
    task=TaskType.CLASSIFICATION,
    seed=13,
    train_batch_size=1,  #for EfficientAD this must be 1 !!! 
    train_transform=resize_transf, 
    eval_transform=resize_transf
)

datamodule.setup()
model=EfficientAd()

logger = AnomalibTensorBoardLogger("tb_logs", name="efficientAD")

engine = Engine(max_epochs=epochs, task=TaskType.CLASSIFICATION,
                    image_metrics={
                        "Precision": {
                            "class_path": "torchmetrics.Precision",
                            "init_args": {"task": "binary"}
                            },
                        "Recall": {
                            "class_path": "torchmetrics.Recall",
                            "init_args": {"task": "binary"}
                        }, 
                        "Accuracy": {
                            "class_path": "torchmetrics.Accuracy",
                            "init_args": {"task": "binary"}
                        }   
                    },
                    callbacks=[ModelCheckpoint(dirpath='checkpoint',filename='{epoch}-{image_AUROC:.2f}',monitor="image_AUROC",mode="max",
                                               every_n_epochs=5, save_last=True),
                               EarlyStopping(monitor="image_AUROC",mode="max",patience=PATIENCE)],
                    logger=logger)

checkpoint_path=None
engine.fit(datamodule=datamodule, model=model,ckpt_path=checkpoint_path)

**Comment by stevensagaert** [comment]
The error is thrown from the EfficientAD model code

**Comment by stevensagaert** [comment]
The reason I do the resize is because I want higher res than (256,256)

**Comment by alexriedel1** [comment]
Here is how you define custom data transforms: https://anomalib.readthedocs.io/en/latest/markdown/guides/how_to/data/transforms.html#using-custom-transforms-for-training-and-evaluation

**Reply by stevensagaert** [reply]
That's what I do. Doesn't work for EfficientAD.

**Reply by alexriedel1** [reply, marked as answer]
no, it looks like you are not using a `Compose` object. 
```

train_transform = Compose(
    [
        RandomAdjustSharpness(sharpness_factor=0.7, p=0.5),
        RandomHorizontalFlip(p=0.5),
        Resize((256, 256), antialias=True),
        Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
    ],
)
eval_transform = Compose(
    [
        Resize((256, 256), antialias=True),
        Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
    ],
)

datamodule = MVTec(train_transform=train_transform, eval_transform=eval_transform)
datamodule.prepare_data()
datamodule.setup()
```

**Reply by stevensagaert** [reply]
That solved it! Thanks! Didn't know you had to Compose even if you only have one transform.

**Comment by samet-akcay** [comment]
Good to see this is sorted, closing the discussion. Thanks!
